server/handler/status.go

```
@@ -50,44 +61,68 @@ func (h *Status) Handle(ctx context.Context, eventType, deliveryID string, paylo
+}
+
+func (h *Status) processOthers(ctx context.Context, event github.StatusEvent) error {
+	repo := event.GetRepo()
+	ownerName := repo.GetOwner().GetLogin()
+	repoName := repo.GetName()
 	commitSHA := event.GetCommit().GetSHA()
+	installationID := githubapp.GetInstallationIDFromEvent(&event)
 
-	if sender.GetLogin() != h.PullOpts.AppName+"[bot]" {
-		logger.Warn().
-			Str(LogKeyAudit, eventType).
-			Str(LogKeyGitHubSHA, commitSHA).
-			Msgf(
-				"Entity '%s' overwrote status check '%s' to state='%s' description='%s' targetURL='%s'",
-				sender.GetLogin(),
-				event.GetContext(),
-				event.GetState(),
-				event.GetDescription(),
-				event.GetTargetURL(),
-			)
-
-		// must be less than 140 characters to satisfy GitHub API
-		desc := fmt.Sprintf("'%s' overwrote status to '%s'", sender.GetLogin(), event.GetState())
-
-		// unlike in other code, use a single context here because we want to
-		// replace a forged context with a failure, not post a general status
-		// if multiple contexts are forged, we will handle multiple events
-		status := &github.RepoStatus{
-			Context:     event.Context,
-			State:       github.String("failure"),
-			Description: &desc,
-		}
-
-		_, _, err := client.Repositories.CreateStatus(ctx, ownerName, repoName, commitSHA, status)
+	client, err := h.NewInstallationClient(installationID)
+	if err != nil {
 		return err
 	}
 
-	return nil
+	ctx, logger := githubapp.PrepareRepoContext(ctx, installationID, repo)
+
+	// Get all PRs for which the event commit references the head of a PR
+	prs, _, err := client.PullRequests.List(ctx, ownerName, repoName, &github.PullRequestListOptions{Head: commitSHA})
+	if err != nil {
+		return err
+	}
+	logger.Debug().Msgf("Context event is for '%s', found %d PRs", event.GetContext(), len(prs))
+
+	for _, pr := range prs {
+		err = h.Evaluate(ctx, installationID, true, pull.Locator{
+			Owner:  ownerName,
+			Repo:   repoName,
+			Number: pr.GetNumber(),
+			Value:  pr,
+		})
+	}
+	return err
```

As implemented, this will only return the the last error and hide any errors returned by earlier evaluations. One way to solve this is to log the intermediate errors and count the number that occurred. Then, return a new error at the end saying something like `failed to evaluate N pull requests`.